core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedRegionPlanGenerator.scala

```diff
@@ -267,12 +278,19 @@ class CostBasedRegionPlanGenerator(
           ) // Edges in chain with blocking edges should not be materialized
         }
 
-        val unvisitedNeighborStates = candidateEdges
+        var unvisitedNeighborStates = candidateEdges
```

Comment:
I suggest you keep using val for functional programming. use a new variable to contain the "filteredStates" rather than replacing the same variable.

Reply:
done